Canceling download of `.tar.gz` leaves bad archive and panics

This occurs if you <kbd>Ctrl</kbd>+<kbd>C</kbd> while a package is downloading.

Example:

```
$ rm -rf ~/.tea/tea.xyz/var/www/llvm.org-* ~/.tea/llvm.org/

$ tea +llvm.org
llvm.org=15.0.6 downloading 862 MiB 3%
^C

$ tea +llvm.org
llvm.org=15.0.6 cache: hit

gzip: stdin: unexpected end of file
tar: Unexpected EOF in archive
tar: Unexpected EOF in archive
tar: Error is not recoverable: exiting now

panic: spilt tea. we’re sorry and we’ll fix it… but you have to report the bug!

...
```

<details>
<summary>Attachment</summary>

```
----------------------------------------------------->> attachment begin
TypeError: Cannot convert object to primitive value
    at install (file:///tmp/77574ee0/src/prefab/install.ts:92:55)
    at async install (file:///tmp/77574ee0/src/hooks/useExec.ts:117:25)
    at async default (file:///tmp/77574ee0/src/hooks/useExec.ts:79:38)
    at async run (file:///tmp/77574ee0/src/app.main.ts:32:63)
    at async file:///tmp/77574ee0/src/app.ts:11:5
<<----------------------------------------------------- attachment end
```

</details>

--- a/src/prefab/install.ts
+++ b/src/prefab/install.ts
@@ -101,7 +101,9 @@ export default async function install(pkg: Package, logger?: Logger): Promise<In
     log_install_msg(install)
 
     return install
-
+  } catch (err) {
+    tarball.rm()  //FIXME resumable downloads!
+    throw err
   } finally {
     await Deno.funlock(rid)
     Deno.close(rid)  // docs aren't clear if we need to do this or not

